deps/rabbitmq_amqp1_0/src/rabbit_amqp1_0_reader.erl

```diff
@@ -435,13 +439,30 @@ handle_1_0_connection_frame(_Frame, State) ->
     maybe_close(State#v1{connection_state = closing}).
 
 handle_1_0_session_frame(Channel, Frame, State) ->
-    case get({channel, Channel}) of
-        {ch_fr_pid, SessionPid} ->
+    case find_session_pid_for_channel(Channel, State) of
```

Comment:
if you use a map for the channels you don't need this function, just use pattern matching

Reply:
I probably misunderstood you. I thought you suggested me to use list rather than map given that there will be very few sessions. But I can change it to map and it  will be simpler for sure. Shall I change it?